sdk/pinocchio/src/pubkey.rs

```diff
@@ -232,3 +234,49 @@ pub fn checked_create_program_address(
 
     create_program_address(seeds, program_id)
 }
+
+/// Derive a Pubkey from another Pubkey, seed, and a program id.
+#[inline]
+pub fn create_with_seed(
+    base: &Pubkey,
+    seed: &[u8],
+    owner: &Pubkey,
+) -> Result<Pubkey, ProgramError> {
+    if seed.len() > MAX_SEED_LEN {
+        return Err(ProgramError::MaxSeedLengthExceeded);
+    }
+
+    if owner.len() >= PDA_MARKER.len() {
+        let slice = &owner[owner.len() - PDA_MARKER.len()..];
+        if slice == PDA_MARKER {
+            return Err(ProgramError::IllegalOwner);
+        }
+    }
+
+    #[cfg(target_os = "solana")]
+    {
+        let mut bytes = core::mem::MaybeUninit::<[u8; PUBKEY_BYTES]>::uninit();
+
+        let vals = &[base, seed, owner];
+
+        let result = unsafe {
+            crate::syscalls::sol_sha256(
+                vals as *const _ as *const u8,
+                vals.len() as u64,
+                bytes.as_mut_ptr() as *mut _,
+            )
+        };
```

Comment:
The syscall always returns `0` when it succeed; otherwise it aborts the vm. There is no need to handle the return.
```suggestion
        unsafe {
            crate::syscalls::sol_sha256(
                vals as *const _ as *const u8,
                vals.len() as u64,
                bytes.as_mut_ptr() as *mut _,
            );
        }
```